File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program107.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and accept one character. 
//		Return frequency of that character.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt,iCount;
	char ch;

	Solution(String str,char ch)
	{
		this.str=str;
		this.iCnt=0;
		this.iCount=0;
		this.ch=ch;
	}

	int CountChar()
	{
		
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)==ch)
			{
				iCount++;
			}
			iCnt++;
		}
		
		return iCount;
	}
}

class Program
{
	public static void main(String args[])
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		System.out.println("Enter the character: ");
		char ch = sObj.next().charAt(0);

		Solution obj = new Solution(str,ch);
		iRet = obj.CountChar();

		System.out.println("Frequency is: "+iRet);
	}
}

/*
OUTPUT:
Enter the string: 
Vishal Patil
Enter the character: 
a
Frequency is: 2

*/

File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program107.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and accept one character. 
//		Return frequency of that character.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCount;
		char ch;

	public:

		Solution(string str,char ch,int iCnt=0,int iCount=0)
		{
			this->str=str;
			this->ch=ch;
		}

		int CountChar()
		{
			while(iCnt!=str.length())
			{
				if(str[iCnt]==ch)
				{
					iCount++;
				}

				iCnt++;
			
			}	
			return iCount;
		}
};

int main()
{
	string str;
	char ch='\0';
	int iRet=0;
	
	cout<<"Enter string: ";
	getline(cin,str);

	cout<<"Enter character: ";
	cin>>ch;
	
	Solution *sObj = new Solution(str,ch);
	iRet = sObj->CountChar();
	
	cout<<"Count is: "<<iRet<<"\n";

	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: Marvellous Multi Os
Enter character: M
Count is: 2

*/
